sympy/integrals/laplace.py

```diff
@@ -1158,15 +1158,14 @@ def _inverse_laplace_apply_simple_rules(f, s, t):
         ma = _F.match(s_dom)
         if ma:
             try:
-                c = check.xreplace(ma)
+                if check.xreplace(ma):
+                    debug('_inverse_laplace_apply_simple_rules match:')
+                    debugf('      f:    %s', (f,))
+                    debugf('      rule: %s o---o %s', (s_dom, t_dom))
+                    debugf('      ma:   %s', (ma,))
+                    return Heaviside(t)*t_dom.xreplace(ma).subs({t_: t}), S.true
             except TypeError:
                 continue
-            if c:
-                debug('_inverse_laplace_apply_simple_rules match:')
-                debugf('      f:    %s', (f,))
-                debugf('      rule: %s o---o %s', (s_dom, t_dom))
-                debugf('      ma:   %s', (ma,))
-                return Heaviside(t)*t_dom.xreplace(ma).subs({t_: t}), S.true
```

Comment:
The change here is so that we move the `if c` inside the try block but it would be better just to change it to `if c is S.true:`. We should not write code that unnecessarily raises TypeError just to catch it again. I understand the need to put a try/except around the `check.xreplace` because we can't avoid the possibility that this substitution might raise a TypeError in some cases but we don't need to put the `if c` inside the try block because we can just test that it is `S.true` explicitly and not raise and catch an exception. Catching `TypeError` in particular is a bad practise because lots of bugs can cause `TypeError` to be raised and so there is the possibility that it can hide bugs. When it is necessary to catch a `TypeError` then we can but the scope needs to be limited as much as possible.

Reply:
Thanks, I was not aware of the huge difference between `if c:` and `if c is S.true:`